### PR TITLE
Add Momentum delivery archetype: ramp-enabled damage tally (#1428)

### LINKED ISSUE #1428: Delivery archetype: Momentum (in-battle escalation)
Delivery archetype from spike #1398 (see [`docs/spikes/1398-utility-in-proficiency-system.md`](docs/spikes/1398-utility-in-proficiency-system.md) → *Follow-on refinement*, and `docs/content-design.md` §4). A **committed, ready-to-implement** engine feature — it follows the opt-in-multiplicative template (#1425) and the **normalized-marginal tally** whose worked reference is the crit migration #1448.

## Fantasy

Damage that **escalates within a battle** — a weak opener that ramps to a strong finish. Distinct from crit's front-loaded spike and DoT's flat attrition, and thematically the answer to a timeout / Zone-3 boss (it closes fights that drag).

## Shape (per the template)

- **Enabler (opt-in):** a ramp effect on gear/skills — mechanically a **stacking self-buff** to a damage/amplification attribute applied on each hit. Expressible on today's effect system (magnitudes sum, expiry refreshes, so a faster build holds higher stacks).
- **Technique path:** *Momentum* — a multiplicative proficiency scaling the ramp, inert (`0`) for a build with no ramp enabler.
- **Tallied magnitude:** the **ramp bonus dealt** — the damage attributable to the stacks above the un-ramped baseline. Momentum is an **overlay** like the others, so it uses the shared tally shape (below); needs a new `BattleStats` tally + `EActivityKey` + `ActivityKeys` binding.

## Attribution — the shared overlay tally shape (settled)

Momentum's ramp is an outgoing amplifier, so it fits the same normalized-marginal tally as crit/Hex/Sunder, settled in [spike #1398 → *Follow-on refinement*](docs/spikes/1398-utility-in-proficiency-system.md#follow-on-refinement--the-overlay-tally-shape-normalized-marginal) and implemented first as the crit reference #1448: the **marginal** ramp bonus (not the full ramped hit), **add-one-in** against the un-ramped baseline (not leave-one-out, so it composes when other overlays stack), as **absolute** enabled damage with the concave normalization on the ramp magnitude. (Being near the top of the chain like crit, Momentum is the least entangled of the overlays — but it follows the same rule for consistency.)

## Dependencies / sequencing

- **Depends on the crit reference #1448** for the tally pattern.
- **Not gated on content** — the mechanic is buildable now; **content placement** (which zone/challenge introduces a ramp enabler) follows the arc as a separate authoring step.

## Notes

- Accrual is off the tick loop (no parity surface); the ramp **effect** itself is battle logic and must stay frontend↔backend consistent.
- Pairs naturally with fast/tempo builds — they stack ramp faster (the afflictor/stacker identity).

Part of #1398.

### PR DESCRIPTION
**Pull request by ginderjeremiah**
## Summary

Implements Momentum (#1428), the third [delivery archetype](docs/content-design.md#delivery-archetypes--the-commitment-rule) from spike #1398 — in-battle escalation via a stacking self-buff.

Per the [spike's follow-on refinement](docs/spikes/1398-utility-in-proficiency-system.md#follow-on-refinement--the-overlay-tally-shape-normalized-marginal), Momentum's enabler ("a ramp effect... mechanically a stacking self-buff to a damage/amplification attribute") is **expressible on today's effect system with no new mechanism** — `Battler.ApplyEffect`'s existing stacking/refresh rules (magnitudes sum, expiry refreshes on re-application) already give a faster-casting build higher ramp stacks for free. So the actual engine work here is the **tally**, following the crit reference (#1448) and mirroring Hex (#1427) closely:

- `EActivityKey.Momentum` (appended after `Hex`).
- `BattleStats.MomentumBonusDealt` — the normalized-marginal training signal.
- `Battler.AppliedMomentum(damageType)` — tracks the ramp's *own* contribution to a typed amplification attribute (a new `AttributeEffectStack.MomentumContribution` field, set via `ApplyEffect(tracksMomentum: true)`), isolated from any static (item/base) amplification the attacker already carries — the self-buff analogue of Hex's `AppliedVulnerability`.
- `BattleContext.DamageTarget` books the bonus per direct-hit portion: the live post-amplification hit minus the same hit *without* the ramp's own contribution (`dealt − rawPortion × r`), both carried through the defender's mitigation, discounted by `1/(1 + r)` — the same saturation crit and Hex use, applied to the ramp's own magnitude (add-one-in against a fixed baseline, never leave-one-out, so it composes with crit/Hex without inflating either).
- `ProficiencyRewardService` routes `MomentumBonusDealt` to the `Momentum` activity key.

**One notable difference from Hex, called out in the docs:** Momentum's marginal is *not* flat in the defender's mitigation. Hex's vulnerability-enabled marginal is flat because it operates on the resistance term itself (it cancels out algebraically); Momentum amplifies the *attacker's own output* pre-mitigation, so a tougher defender mitigates the ramp bonus exactly like the rest of the hit — the same property the crit bonus has. This is intentional and mathematically expected, not a bug.

**Scope note:** Momentum's tally is scoped to direct hits, matching crit's precedent (crit's bonus also never applies to DoT ticks). A DoT's frozen-at-cast-time amplification already includes any active ramp for free (via the existing `ApplySkillEffect`/`AmplifyDamage` freeze), but the *tally* doesn't extend to DoT ticks — extending it would require re-deriving live per-tick attribution the way Hex does for its (defender-side, live-sampled) vulnerability, which doesn't have a clean analogue here since the amplification is frozen at cast, not sampled live.

**Content authoring** (which zone/skill introduces a ramp enabler, and the Momentum Technique path itself) is explicitly out of scope per the spike — `content/paths.json` ships empty today, same as Hex's precedent.

## Changes

- `Game.Core/Enums.cs` — `EActivityKey.Momentum`.
- `Game.Core/Attributes/DamageTypes.cs` — `IsAmplificationAttribute`, the Momentum-enabler gate (mirrors `IsResistanceAttribute`).
- `Game.Core/Battle/Battler.cs` — `AppliedMomentum`, `AttributeEffectStack.MomentumContribution`, `ApplyEffect(tracksMomentum)`.
- `Game.Core/Battle/BattleContext.cs` — momentum-tracking gate in `ApplySkillEffect`; per-portion booking in `DamageTarget` (refactored `AmplifiedPortion` to expose the shared raw-portion split via a new `PortionRawDamage` helper, needed for the un-ramped counterfactual).
- `Game.Core/Battle/BattleStats.cs` — `MomentumBonusDealt`.
- `Game.Application/Services/ProficiencyRewardService.cs` — wires the new activity key.
- `UI/src/lib/api/types/enums.ts` — regenerated via `dotnet run --project Game.Api -- codegen`.
- `UI/src/routes/admin/workbench/progression/progression-helpers.ts` — Workbench label for the new activity key.
- `docs/backend-battle.md`, `docs/game-design.md` — document the tally, its formula, and the flatness difference from Hex.

## Test plan

- [x] `dotnet build` — clean, 0 warnings/errors.
- [x] `dotnet test` (all 4 backend suites) — 2638 tests passed, 0 failed.
- [x] New `Game.Core.Tests/Battle/BattleContextTests.cs` coverage: no-ramp, static-amplification-doesn't-train, applied-ramp marginal math, heavy-investment saturation, mitigation interaction (the flatness difference from Hex), crit+Momentum independence, expiry, and player-only isolation.
- [x] New `Game.Application.Tests/Services/ProficiencyRewardServiceTests.cs` coverage: `MomentumBonus_TrainsTheMomentumPath`, `NoMomentumBonus_TrainsNoMomentum`.
- [x] `dotnet format --verify-no-changes` — clean.
- [x] `npm run lint` (ESLint + svelte-check) — clean.
- [x] `npm run test:unit:ci` (full frontend suite) — passed, coverage thresholds met.
- [x] Frontend codegen drift check — regenerated `enums.ts` and diffed only the expected `Momentum = 31` addition.

Closes #1428

---
_Generated by [Claude Code](https://claude.ai/code/session_01NQPNvCxdStRU8Hzs7dfQU5)_

### CODE REVIEW

File: Game.Core/Battle/Battler.cs [hunk 1]
```diff
@@ -222,6 +222,46 @@ private double StackVulnerabilityContribution(EAttribute attribute)
...
+            }
+
+            return contribution > 0 ? contribution : 0;
+        }
+
+        // The ramp-tracked amplification this battler has applied to one of its own attributes, or 0 when no
+        // effect stack for it is active. A linear scan over the affected-attribute count, like GetOrCreateStack.
+        private double StackMomentumContribution(EAttribute attribute)
+        {
+            foreach (var stack in _attributeStacks!)
```
Comment (ginderjeremiah) [inline Game.Core/Battle/Battler.cs:254]: Nit (non-blocking): the `_attributeStacks!` null-forgiving operator contradicts CLAUDE.md's "AVOID the null-forgiving operator" rule. It's safe here — `AppliedMomentum` null-guards before calling — and it faithfully mirrors the existing `StackVulnerabilityContribution`, so it's a pre-existing-pattern deviation rather than something to change in this PR. Tracked for both sibling helpers in follow-up #1460.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by ginderjeremiah — COMMENTED** [review]
## Review: Momentum delivery archetype — ramp-enabled damage tally (#1428)

**Verdict:** Approve _(submitted as a Comment because I'm the PR author; GitHub won't accept a gating review on one's own PR)_ — correct, well-tested, and a faithful copy of the established Hex/crit overlay-tally pattern.

I verified the core attribution math end-to-end:

- `AmplifyDamage` is purely additive (`raw × (1 + Σ amplification)`), so the counterfactual `unrampedDealt = dealt − rawPortion × r` removes **exactly** the ramp's own contribution and nothing else — including for multi-attribute types, since `AppliedMomentum` sums the tracked contribution across the type's amplification attributes. ✔
- `MomentumContribution` is fed the same scaled `amount` that's folded into the additive combined modifier (`ApplyEffect`), so the tracked `r` matches the ramp's real effect on `dealt`. ✔
- Both hits run through the same `ComputeNetDamage`, and the `/(1 + r)` saturation matches crit/Hex. The "not flat in defender mitigation" property (vs. Hex) is correct and intentional — Momentum amplifies the attacker's output, like crit. ✔
- Player-only booking is correct: the tally lives only in the `_isPlayerActive` branch and reads `_activeBattler.AppliedMomentum`, so an enemy self-ramp never trains the player. ✔
- The `tracksMomentum` gate (Self + Additive + amplification attribute) is mutually exclusive with `tracksVulnerability` (Opponent + Additive + resistance attribute), so no double-tracking. ✔

Backend-only tally with no parity surface — consistent with Hex and crit (confirmed no overlay tally exists in the frontend sim), so the ramp *effect* stays mirrored via the existing effect system while the *training signal* correctly stays backend-only. The direct-hits-only scope (no DoT tick tally) matches crit's precedent and is clearly documented.

### Nits

- `Battler.cs:254` — `_attributeStacks!` null-forgiving operator contradicts CLAUDE.md's rule to avoid `!`. Safe here (caller null-guards) and it mirrors the existing `StackVulnerabilityContribution`, so it's a pre-existing-pattern deviation. Filed as tech-debt follow-up #1460 covering both sibling helpers.

### Test coverage

Strong and matches the project's rules:
- `BattleContextTests` — 8 new cases: no-ramp, static-amplification-doesn't-train, applied-ramp marginal, heavy-investment saturation, the mitigation-interaction (flatness difference from Hex), crit+Momentum independence, expiry, and enemy-attacking isolation. Covers the edge cases that matter.
- `ProficiencyRewardServiceTests` — routing on/off (`MomentumBonus_TrainsTheMomentumPath`, `NoMomentumBonus_TrainsNoMomentum`).
- Frontend workbench label test added.
- No frontend battle-step mirror needed — the tally is backend-only (no parity surface), same as Hex/crit.

### What's good

- The `PortionRawDamage` extraction is the right refactor — it keeps the player/enemy loops and the un-ramped counterfactual on one shared split so they can't diverge, without leaving `AmplifiedPortion` dead (still used on the enemy branch).
- Documentation is thorough and honest, explicitly calling out the flatness difference from Hex and the DoT-scope decision rather than glossing over them.
- Faithful adherence to the reference tally shape keeps the four overlays (crit/Hex/Momentum/Cull) consistent.

---
_Generated by [Claude Code](https://claude.ai/code)_
